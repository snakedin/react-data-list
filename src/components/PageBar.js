import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withOptions } from "../options/withOptions";
import Paginator from "./pager/Paginator";
import Summary from "./pager/Summary";
import SizeSwitcher from "./pager/SizeSwitcher";
import PageSwitcher from "./pager/PageSwitcher";

class PageBar extends Component {

    static propTypes = {
        page: PropTypes.number,
        pageSize: PropTypes.number,
        totalCount: PropTypes.number,
        lastPage: PropTypes.bool,
        onParamChanged: PropTypes.func
    };

    /**
     * Render element
     */
    render() {

        const {
            totalCount, lastPage, page, pageSize, onParamChanged,
            options: { showPageBar, renderPageBar }
        } = this.props;

        if (showPageBar === false || totalCount === 0) {
            return '';
        }

        let paginator;
        let summary;

        // If lastPage param was set, we will use Prev/Next mode instead of classic pagination
        if (lastPage !== undefined && lastPage !== null && typeof lastPage === "boolean") {
            paginator = <PageSwitcher page={ page } lastPage={ lastPage } onParamChanged={ onParamChanged } />;
        } else {
            paginator = <Paginator page={ page } pageSize={ pageSize } totalCount={ totalCount } onParamChanged={ onParamChanged } />;
            summary = <Summary page={ page }  pageSize={ pageSize } totalCount={ totalCount }/> ;
        }

        const sizeSwitcher = <SizeSwitcher pageSize={ pageSize } onParamChanged={ onParamChanged }/>;

        if (typeof renderPageBar === 'function') {
            return renderPageBar(paginator, summary, sizeSwitcher);
        } else {
            return (
                <div className="row align-items-center mb-3 no-gutters">
                    <div className="col-12 col-lg-4 order-2 order-lg-1 text-left mb-3">{ paginator }</div>
                    <div className="col-12 col-lg-4 order-1 order-lg-2 text-center mb-3">{ summary }</div>
                    <div className="col-2 offset-5 col-lg-1 order-3 offset-lg-3 text-right mb-3">{ sizeSwitcher }</div>
                </div>
            );
        }
    }
}

export default withOptions(PageBar);