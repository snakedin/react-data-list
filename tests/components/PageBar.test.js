import React from "react";
import { render } from "@testing-library/react";
import "../mocks/withOptionsMock";
import PageBar from "../../src/components/PageBar";

it("PageBar render", () => {

  const { asFragment, rerender } = render(<PageBar page={25} lastPage={false}/>);
  expect(asFragment()).toMatchSnapshot('short prev/next page switcher');

  rerender(<PageBar page={13} pageSize={10} totalCount={500}/>);
  expect(asFragment()).toMatchSnapshot('full pagination');
});
