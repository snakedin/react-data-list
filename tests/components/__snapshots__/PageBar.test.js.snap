// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageBar render: full pagination 1`] = `
<DocumentFragment>
  <div
    class="row align-items-center mb-3 no-gutters"
  >
    <div
      class="col-12 col-lg-4 order-2 order-lg-1 text-left mb-3"
    >
      <ul
        class="pagination mb-0 justify-content-center justify-content-lg-start"
      >
        <li
          class="page-item"
        >
          <button
            aria-label="Previous"
            class="page-link"
          >
            <span
              aria-hidden="true"
            >
              «
            </span>
          </button>
        </li>
        <li
          class="page-item"
        >
          <button
            class="page-link"
          >
            8
          </button>
        </li>
        <li
          class="page-item"
        >
          <button
            class="page-link"
          >
            9
          </button>
        </li>
        <li
          class="page-item"
        >
          <button
            class="page-link"
          >
            10
          </button>
        </li>
        <li
          class="page-item"
        >
          <button
            class="page-link"
          >
            11
          </button>
        </li>
        <li
          class="page-item"
        >
          <button
            class="page-link"
          >
            12
          </button>
        </li>
        <li
          class="page-item active"
        >
          <button
            class="page-link"
          >
            13
          </button>
        </li>
        <li
          class="page-item"
        >
          <button
            class="page-link"
          >
            14
          </button>
        </li>
        <li
          class="page-item"
        >
          <button
            class="page-link"
          >
            15
          </button>
        </li>
        <li
          class="page-item"
        >
          <button
            class="page-link"
          >
            16
          </button>
        </li>
        <li
          class="page-item"
        >
          <button
            class="page-link"
          >
            17
          </button>
        </li>
        <li
          class="page-item"
        >
          <button
            aria-label="Next"
            class="page-link"
          >
            <span
              aria-hidden="true"
            >
              »
            </span>
          </button>
        </li>
      </ul>
    </div>
    <div
      class="col-12 col-lg-4 order-1 order-lg-2 text-center mb-3"
    >
      <span>
        Showing 
        <b>
          121-130
        </b>
         of 
        <b>
          500
        </b>
         items.
      </span>
    </div>
    <div
      class="col-2 offset-5 col-lg-1 order-3 offset-lg-3 text-right mb-3"
    >
      <select
        class="form-control"
      >
        <option
          value="10"
        >
          10
        </option>
        <option
          value="20"
        >
          20
        </option>
        <option
          value="30"
        >
          30
        </option>
      </select>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PageBar render: short prev/next page switcher 1`] = `
<DocumentFragment>
  <div
    class="row align-items-center mb-3 no-gutters"
  >
    <div
      class="col-12 col-lg-4 order-2 order-lg-1 text-left mb-3"
    >
      <ul
        class="pagination mb-0 justify-content-center justify-content-lg-start"
      >
        <li
          class="page-item"
        >
          <button
            aria-label="Previous"
            class="page-link"
          >
            <span
              aria-hidden="true"
            >
              «
            </span>
          </button>
        </li>
        <li
          class="page-item"
        >
          <button
            aria-label="Next"
            class="page-link"
          >
            <span
              aria-hidden="true"
            >
              »
            </span>
          </button>
        </li>
      </ul>
    </div>
    <div
      class="col-12 col-lg-4 order-1 order-lg-2 text-center mb-3"
    />
    <div
      class="col-2 offset-5 col-lg-1 order-3 offset-lg-3 text-right mb-3"
    >
      <select
        class="form-control"
      >
        <option
          value="10"
        >
          10
        </option>
        <option
          value="20"
        >
          20
        </option>
        <option
          value="30"
        >
          30
        </option>
      </select>
    </div>
  </div>
</DocumentFragment>
`;
